riscv/decode_macros.h

```diff
@@ -23,11 +23,11 @@
 #define RS3 READ_REG(insn.rs3())
 #define WRITE_RD(value) WRITE_REG(insn.rd(), value)
 
-#ifndef RISCV_ENABLE_COMMITLOG
+#if defined(DECODE_MACRO_USAGE_FAST)
 # define WRITE_REG(reg, value) ({ CHECK_REG(reg); STATE.XPR.write(reg, value); })
 # define WRITE_FREG(reg, value) DO_WRITE_FREG(reg, freg(value))
 # define WRITE_VSTATUS {}
-#else
+#elif defined(DECODE_MACRO_USAGE_LOGGED)
```

Comment:
Add an `#else [die]` if neither macro is defined. Or just use a single macro and `#ifdef`/`#else`.

Reply:
I prefer `#else [die]`, but a few other non-instruction sources rely on `decode_macros.h`. This structure enforces that only the instruction templates can use the `WRITE_REG` macros.

A future PR should enforce that only the instruction templates include `decode_macros.h`